python/gstaichi/lang/ast/ast_transformers/function_def_transformer.py

```diff
@@ -274,15 +334,23 @@ def _transform_func_arg(
 
     @staticmethod
     def _transform_as_func(ctx: ASTTransformerContext, node: ast.FunctionDef, args: ast.arguments) -> None:
+        assert ctx.argument_data is not None
+        assert ctx.func is not None
         for data_i, data in enumerate(ctx.argument_data):
-            argument = ctx.func.arguments[data_i]
+            argument = ctx.func.arg_metas[data_i]
             FunctionDefTransformer._transform_func_arg(
                 ctx,
                 argument.name,
                 argument.annotation,
                 data,
```

Comment:
Same, either use keyword arguments for clarity, or keep all input arguments in one line if not important.